src/main/java/com/arpnetworking/metrics/mad/model/DefaultRecord.java

```diff
@@ -28,6 +28,10 @@
 /**
  * Default implementation of the <code>Record</code> interface.
  *
+ * <b>IMPORTANT:</b> Instances are only hashed and compare based on their
+ * identifier, and are considered interchangeable if their identifiers match
+ * irregardless of their content.
+ *
```

Comment:
I added this note because it caught me by surprise. I don't know if record instances are compared anywhere at the moment. I am supportive of changing the default compare+hash to be more standard and fixing anywhere that relies on this id comparison. Filed issue: #196